### PR TITLE
Builder postprocess

### PR DESCRIPTION
**Pull request by kinnison**
This work is to enable work to complete https://github.com/gtk-rs/gtk/issues/984 by allowing us to register something in `gtk::ApplicationBuilder::build()`

### CODE REVIEW

**Comment by EPashkin** [comment]
@kinnison Thanks, LGFM

File: README.md [hunk 1]
```diff
@@ -18,13 +18,13 @@ Keep it in mind that since `gir` is still under development, it generates warnin
 
 ## `gir` Modes
 
-There are two main modes of generation for `gir`; *FFI* and *API*.
+There are two main modes of generation for `gir`; _FFI_ and _API_.
```
Comment (GuillaumeGomez) [inline README.md:21]: You don't like stars? :o

File: README.md [hunk 2]
```diff
@@ -147,6 +147,20 @@ status = "generate"
 generate_builder = true
 ```
 
+If there is some work which has to be done post-construction before the builder's
```
Comment (GuillaumeGomez) [inline README.md:150]: Nice! :+1:

File: README.md (hunk 1 above)
Comment (kinnison) [reply README.md:21]: This is a side-effect of me not turning my markdown formatter off in my editor.  I can clean these up if you'd prefer.

**Comment by kinnison** [comment]
The CI is green.  I can push an "undoing" of the messup in README, which would leave the same commits for the code, but would require CI to run again since it cannot distinguish code from README for this kind of thing.  Whichever you would prefer I can do.  Just let me know.

Once this is merged, I will update the submodule in gtk-rs/gtk and then sort out a PR using this feature.

**Comment by GuillaumeGomez** [comment]
I guess it's good to go @EPashkin ?

**Comment by sdroege** [comment]
Looks good to me too, at least the generated code

**Comment by GuillaumeGomez** [comment]
@EPashkin Since you have already approved, I'll merge this PR to let @kinnison move forward on this.

**Pull request merged**